x/records/keeper/deposit_record.go

```diff
@@ -75,6 +76,12 @@ func (k Keeper) GetDepositRecord(ctx sdk.Context, id uint64) (val types.DepositR
 func (k Keeper) RemoveDepositRecord(ctx sdk.Context, id uint64) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.DepositRecordKey))
 	store.Delete(GetDepositRecordIDBytes(id))
+	count := k.GetDepositRecordCount(ctx)
+	if count > 0 {
```

Comment:
flagging as a bug (per offline discussion) that we should not merge in

Reply:
reverted, ty